feg/gateway/services/n7_n40_proxy/servicers/session_controller.go

```diff
@@ -94,6 +95,22 @@ func (srv *CentralSessionController) TerminateSession(
 	ctx context.Context,
 	request *protos.SessionTerminateRequest,
 ) (*protos.SessionTerminateResponse, error) {
-
-	return (&protos.UnimplementedCentralSessionControllerServer{}).TerminateSession(ctx, request)
+	glog.V(2).Info("Deleting session")
+	if err := validateSessionTerminateRequest(request); err != nil {
+		return nil, err
+	}
+	smPolicyId := n7.GetSmPolicyId(request.GetTgppCtx())
+	if len(smPolicyId) == 0 {
+		return nil, fmt.Errorf("SmPolicyId not found in the SessionTerminateResponse TgppCtx")
+	}
```

Comment:
why don't you make `GetSmPolicyId` to retunr error in case len == 0? This way the error generation is encapsulated  

```
	smPolicyId, err := n7.GetSmPolicyId(request.GetTgppCtx())
	if err =! nil {
		err = fmt.Errorf("SessionTerminateRequest failed due to missing policyId: %s", err)
		glog.Error(err)
		return nil, err
	}
```

Reply:
Done